src/components/TextInputComponent/TextInputComponent.css

```diff
@@ -0,0 +1,49 @@
+.text-input-component-container {
+	display: flex;
+	flex-direction: column;
+	gap: var(--spacing-xs);
+
+	.input-label-container {
+		align-items: center;
+		display: flex;
+		justify-content: space-between;
+
+		& > label {
+			color: var(--color-foreground);
+			display: flex;
+			flex-grow: 1;
+			font-size: var(--text-sm);
+			font-weight: var(--weight-semibold);
+			justify-content: space-between;
+		}
+	}
+
+	& > input {
+		background-color: var(--color-card);
+		border: var(--border);
+		border-radius: var(--rounded-xs);
+		gap: var(--spacing-xs);
+		height: auto;
+		padding: var(--spacing-xs);
+		width: 100%;
+
+		&[data-error='true'] { border-color: var(--color-error); }
+
+		&[aria-disabled='true'] {
+			background-color: var(--color-background);
+			cursor: not-allowed;
+			pointer-events: none;
+		}
+	}
+
+	&:has(input:required) > span.input-label-container > label::after {
```

Comment:
Nitpick:

Is there a reason why we're switching between a pseudo-class select and a full attribute selector?

There's no functional difference.  It hinders readability and consistency.

Reply:
@madcampos he's using two different ways for doing the SAME thing.  Why?